src/main/java/com/bandwidth/sdk/BandwidthClient.java

```diff
@@ -478,4 +512,92 @@ protected URI buildUri(final String path, final List<NameValuePair> queryStringP
             throw new IllegalStateException("Invalid uri", e);
         }
     }
+
+    private HttpClient createHttpClient( final String maxTotal,
+                                         final String defaultMaxPerRoute) {
+        // Following recommendations from
+        // https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+
+        cm.setMaxTotal(Integer.parseInt(maxTotal));
+        cm.setDefaultMaxPerRoute(Integer.parseInt(defaultMaxPerRoute));
+
+        HttpClient httpClient = HttpClients.custom()
+                .setConnectionManager(cm)
+                .setKeepAliveStrategy(getStrategy())
+                .setBackoffManager(new AIMDBackoffManager(cm))
+                .build();
+        this.idleConnectionMonitorThread = new IdleConnectionMonitorThread(cm);
```

Comment:
I would change the name from idleConnectionMonitorThread to idleConnectionMonitorRunnable, because it is a runnable